definitions/procedures/backup/online/candlepin_db.rb

```diff
@@ -0,0 +1,19 @@
+module Procedures::Backup
+  module Online
+    class CandlepinDB < ForemanMaintain::Procedure
```

Comment:
```
Backup Candlepin database online:
- Getting Candlepin DB dump                                           [FAIL]
undefined local variable or method `command' for candlepin_database<Features::CandlepinDatabase>:Features::CandlepinDatabase
--------------------------------------------------------------------------------

```
```
E, [2018-04-17 09:23:26-0400 #11939] ERROR -- : undefined local variable or method `command' for candlepin_database<Features::CandlepinDatabase>:Features::CandlepinDatabase (NameError)
/root/foreman_maintain/lib/foreman_maintain/concerns/base_database.rb:121:in `base_command'
/root/foreman_maintain/lib/foreman_maintain/concerns/base_database.rb:130:in `dump_command'
/root/foreman_maintain/lib/foreman_maintain/concerns/base_database.rb:42:in `dump_db'
/root/foreman_maintain/definitions/procedures/backup/online/candlepin_db.rb:14:in `block in run'
/root/foreman_maintain/lib/foreman_maintain/reporter/cli_reporter.rb:135:in `with_spinner'
/root/foreman_maintain/definitions/procedures/backup/online/candlepin_db.rb:13:in `run'
/root/foreman_maintain/lib/foreman_maintain/executable.rb:139:in `__run__'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:83:in `block (2 levels) in run'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:107:in `capture_errors'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:82:in `block in run'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:101:in `with_metadata_calculation'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:81:in `run'
/root/foreman_maintain/lib/foreman_maintain/runner.rb:113:in `run_step'
/root/foreman_maintain/lib/foreman_maintain/runner.rb:102:in `run_steps'
/root/foreman_maintain/lib/foreman_maintain/runner.rb:95:in `execute_scenario_steps'
/root/foreman_maintain/lib/foreman_maintain/runner.rb:48:in `run_scenario'
/root/foreman_maintain/lib/foreman_maintain/runner.rb:31:in `block in run'
/root/foreman_maintain/lib/foreman_maintain/runner.rb:30:in `each'
/root/foreman_maintain/lib/foreman_maintain/runner.rb:30:in `run'
/root/foreman_maintain/lib/foreman_maintain/cli/base.rb:47:in `run_scenario'
/root/foreman_maintain/lib/foreman_maintain/cli/backup_command.rb:38:in `perform_backup'
/root/foreman_maintain/lib/foreman_maintain/cli/backup_command.rb:86:in `execute'
/usr/share/gems/gems/clamp-1.0.0/lib/clamp/command.rb:68:in `run'

```

Reply:
@ntkathole I don't see how this could happen, could you re-try with updated code please?

Reply:
@mbacovsky i tried with updated codebase, step Backup Candlepin database online passed but Compress backup data to save space failed with undefined local variable or method `pg_tar' for #<Procedures::Backup::CompressData:0x000000014a0a00>

E, [2018-04-17 11:06:08-0400 #22851] ERROR -- : undefined local variable or method `pg_tar' for #<Procedures::Backup::CompressData:0x000000014a0a00
> (NameError)
/root/foreman_maintain/definitions/procedures/backup/compress_data.rb:18:in `compress_file'
/root/foreman_maintain/definitions/procedures/backup/compress_data.rb:10:in `run'
/root/foreman_maintain/lib/foreman_maintain/executable.rb:139:in `__run__'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:83:in `block (2 levels) in run'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:107:in `capture_errors'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:82:in `block in run'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:101:in `with_metadata_calculation'
/root/foreman_maintain/lib/foreman_maintain/runner/execution.rb:81:in `run'

Reply:
Was this comment resolved?

Reply:
Seems so, the new error is later in the execution and was fixed as well. But I let @ntkathole to confirm